File: app/src/main/java/de/sysitprep/app/data/model/Session.kt
package de.sysitprep.app.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import de.sysitprep.app.data.database.Converters

/**
 * Tracks a user's answer to a specific question in a session.
 */
@Entity(tableName = "user_answers")
@TypeConverters(Converters::class)
data class UserAnswer(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val sessionId: Long,
    val questionId: Long,
    val selectedIndices: List<Int>,
    val isCorrect: Boolean,
    val timeSpentSeconds: Int,
    val answeredAt: Long = System.currentTimeMillis()
)

/**
 * A learning or exam session.
 */
@Entity(tableName = "sessions")
@TypeConverters(Converters::class)
data class Session(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val sessionType: SessionType,
    val fachrichtung: Fachrichtung,
    val examType: ExamType? = null,
    val lernfeld: Int? = null,
    val totalQuestions: Int,
    val correctAnswers: Int = 0,
    val durationSeconds: Int = 0,
    val completedAt: Long? = null,
    val startedAt: Long = System.currentTimeMillis(),
    val questionIds: List<Long> = emptyList()
)

enum class SessionType(val displayName: String) {
    LERNFELD("Lernfeld Training"),
    EXAM_SIMULATION("Prüfungssimulation"),
    QUICK_QUIZ("Schnell-Quiz"),
    BOOKMARK_REVIEW("Lesezeichen Wiederholung")
}

File: ExamForge-iOS/ExamForge/Models/Session.swift
import Foundation

// MARK: - SessionType

enum SessionType: String, Codable {
    case LERNFELD
    case EXAM_SIMULATION
    case QUICK_QUIZ
    case BOOKMARK_REVIEW

    var displayName: String {
        switch self {
        case .LERNFELD:         return "Lernfeld Training"
        case .EXAM_SIMULATION:  return "Pr\u{00FC}fungssimulation"
        case .QUICK_QUIZ:       return "Schnell-Quiz"
        case .BOOKMARK_REVIEW:  return "Lesezeichen Wiederholung"
        }
    }
}

// MARK: - Session

struct Session: Identifiable {
    let id: Int64
    let sessionType: SessionType
    let fachrichtung: Fachrichtung
    let examType: ExamType?
    let lernfeld: Int?
    let totalQuestions: Int
    var correctAnswers: Int
    var durationSeconds: Int
    var completedAt: Date?
    let startedAt: Date
    var questionIds: [Int64]
}

// MARK: - UserAnswer

struct UserAnswer: Identifiable {
    let id: Int64
    let sessionId: Int64
    let questionId: Int64
    let selectedIndices: [Int]
    let isCorrect: Bool
    let timeSpentSeconds: Int
    let answeredAt: Date
}
